### PR TITLE
[Bug fix] LPS-146037 Race condition when adding companies concurrently

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test

File: portal-impl/src/com/liferay/portal/util/PortalInstances.java [hunk 1]
```diff
@@ -66,17 +68,7 @@
...
-		}
-
-		long[] companyIds = new long[_companyIds.length + 1];
-
-		System.arraycopy(_companyIds, 0, companyIds, 0, _companyIds.length);
-
-		companyIds[_companyIds.length] = companyId;
-
-		_companyIds = companyIds;
+		_companyIds.addLast(companyId);
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/util/PortalInstances.java:71]: @dantewang Based on this checking "ArrayUtil.contains(_companyIds, companyId)", the elements in the collection should be unique, but Deque can not make sure the entries are unique, let's use ConcurrentSkipListSet.

**Comment by tinatian** [comment]
@dantewang Please see my comment above and resend.

Thanks.
Tina.

**Comment by tinatian** [comment]
ci:close

**Pull request closed without merging**